stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -728,16 +727,19 @@ julia> rank(Matrix(I, 3, 3))
 julia> rank(diagm(0 => [1, 0, 2]))
 2
 
-julia> rank(diagm(0 => [1, 0.001, 2]), 0.1)
+julia> rank(diagm(0 => [1, 0.001, 2]), rtol=0.1)
 2
 
-julia> rank(diagm(0 => [1, 0.001, 2]), 0.00001)
+julia> rank(diagm(0 => [1, 0.001, 2]), rtol=0.00001)
+3
+
+julia> rank(diagm(0 => [1, 0.001, 2]), atol=0.00001, rtol=0.00001)
 3
 ```
 """
-function rank(A::AbstractMatrix, tol::Real = min(size(A)...)*eps(real(float(one(eltype(A))))))
+function rank(A::AbstractMatrix; atol=0.0, rtol=0.0)
```

Comment:
This seems like a breaking change?  We need to keep a `rank(A, tol) = rank(A, rtol=tol)` method in Julia 1.x.

Why is there no default `rtol`?

Reply:
Good point; leave a `TODO` comment about deprecating `tol` in Julia 2.0, however.